src/containers/Validators/index.tsx

```diff
@@ -0,0 +1,229 @@
+  // eslint-disable-next-line camelcase -- from VHS
+  master_key?: string
+  // eslint-disable-next-line camelcase -- from VHS
+  signing_key?: string
+}
+
+const Validator = (props: Props) => {
+  const [reports, setReports] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState(null)
+  const [data, setData] = useState<ValidatorData>({})
+  const [width, setWidth] = useState(0)
+  const ref = useRef<any>(null)
+  const { t, i18n } = useTranslation()
+  const rippledSocket = useContext(SocketContext)
+  const { match } = props
+  const { identifier = '', tab = 'details' } = match.params
+  const { language } = i18n
+
+  useEffect(() => {
+    let short = ''
+    if (data.domain) {
+      short = data.domain
+    } else if (data.master_key) {
+      short = `${data.master_key.substr(0, 8)}...`
+    } else if (data.signing_key) {
+      short = `${data.signing_key.substr(0, 8)}...`
+    }
+    document.title = `Validator ${short} | ${t('xrpl_explorer')}`
+  }, [data.master_key, data.signing_key, data.domain, t])
+
+  useEffect(() => {
+    analytics(ANALYTIC_TYPES.pageview, {
+      title: 'Validator',
+      path: `/validators/:identifier/${tab}`,
+    })
+  }, [tab])
+
+  // TODO: move width calculation to SimpleTab once that's using hooks
+  useEffect(() => {
+    setWidth(ref.current ? ref.current.offsetWidth : 0)
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- actually needed here, not sure why it's complaining
+  }, [ref.current])
+
+  function fetchValidatorReport() {
+    axios.get(`/api/v1/validator_report/${identifier}`).then((resp) => {
+      setReports(resp.data)
+    })
+  }
+
+  function fetchValidatorData() {
+    if (
+      identifier &&
+      identifier !== data.master_key &&
+      identifier !== data.signing_key
+    ) {
+      const url = `/api/v1/validators?verbose=true&key=${identifier}`
+      axios
+        .get(url)
+        .then((response) => {
+          if (!response.data.ledger_hash) {
+            getLedger(response.data.ledger_index, rippledSocket).then(
+              (ledgerData) => {
+                setData({
+                  ...response.data,
+                  ledger_hash: ledgerData.ledger_hash,
+                  last_ledger_time: ledgerData.close_time,
+                })
+                setIsLoading(false)
+              },
+            )
+          } else {
+            setData(response.data)
+            setIsLoading(false)
+          }
+        })
+        .catch((axiosError) => {
+          const status =
+            axiosError.response && axiosError.response.status
+              ? axiosError.response.status
+              : SERVER_ERROR
+          analytics(ANALYTIC_TYPES.exception, {
+            exDescription: `${url} --- ${JSON.stringify(axiosError)}`,
+          })
+          setError(status)
+          setIsLoading(false)
+        })
+    }
+  }
+
+  useQuery(['fetchValidatorData'], async () => fetchValidatorData(), {
```

Comment:
- `isLoading` is provided by `useQuery` so you don't have to manage the state of `loading` yourself.
- `data` is provided and is the resolved result of the promise.
- `error` is provided by `useQuery` as well by the promise returning a rejection

Reply:
I couldn't get this to work for me - there were a bunch of errors